crates/aptos-telemetry-service/src/prometheus_push_metrics.rs

```diff
@@ -86,6 +99,16 @@ pub async fn handle_metrics_ingest(
                 METRICS_INGEST_BACKEND_REQUEST_DURATION
                     .with_label_values(&[name, "Unknown"])
                     .observe(start_timer.elapsed().as_secs_f64());
+                context
+                    .downtime_metrics_cache()
+                    .write()
+                    .push_back(MetricsEntry {
+                        context: context.clone(),
+                        claims: claims.clone(),
+                        encoding: encoding.clone(),
+                        metrics_body: metrics_body.clone(),
+                        timestamp,
+                    });
```

Comment:
Caching here will lead to duplication. We have the ability to push to multiple prometheus backends for high availability (for mainnet, we push to two backends), so if we push to cache on each failure, we will end up caching multiple times. See line 73 which iterates over client vec.

We should instead cache if we weren't able to send to any backends.

Reply:
Thanks for pointing it out. Fixed it.